### ISSUE TITLE
Wrong behavior on pressing different hotkey at the same time

### ISSUE THREAD

**Issue by pionxzh** [JohannesKlauss/react-hotkeys-hook#850]
### Version
4.0.4

### Link to minimal reproduction
https://stackblitz.com/edit/react-p9vi6t?devToolsHeight=33&file=src/App.js

### Steps to reproduce
1. Register hotkey for `UP`, `RIGHT`, `DOWN`, `LEFT`
2. Log the result in your own way
3. Press [`UP`, `RIGHT`] at the same time (`UP` first)
4. See the result / console log

### What is expected?
Press [`UP`, `RIGHT`] at the same time
=> ['UP', 'RIGHT']

### What is actually happening?
Press [`UP`, `RIGHT`] at the same time
=> ['UP', 'UP']

![image](https://user-images.githubusercontent.com/9910706/203126477-50f5854d-ca83-4b32-9380-fa52091bbc47.png)

## Case 2

Try this one, it's even worse 😢 (see console log)

Press [`UP`, `RIGHT`] at the same time
=> ['UP', 'RIGHT', 'UP']

```js
import React, { useState } from 'react';
import { useHotkeys } from 'react-hotkeys-hook';
import './style.css';

export default function App() {
  useHotkeys(
    'UP',
    () => {
      console.log('UP');
    },
    { preventDefault: true }
  );

  useHotkeys(
    'DOWN',
    () => {
      console.log('DOWN');
    },
    { preventDefault: true }
  );

  useHotkeys(
    'LEFT',
    () => {
      console.log('LEFT');
    },
    { preventDefault: true }
  );

  useHotkeys(
    'RIGHT',
    () => {
      console.log('RIGHT');
    },
    { preventDefault: true }
  );

  return (
    <div>
      <h1>react-hotkeys-hook</h1>
      <p>Pressing Arrow Button at the same time</p>
      <p>For example, press [UP, RIGHT] (try UP first)</p>
      <p>Then you will see the wrong pressed result</p>
      <br />
    </div>
  );
}
```

**Comment by JohannesKlauss** [comment]
Thanks for raising the issue, I'll have a look at it

**Comment by psychedelicious** [comment]
I have run into an issue which feels very similar. 

For combination hotkeys which include `meta` e.g. `meta+z` (overriding undo on macOS), if you trigger the hotkey but do not release `meta`, any subsequent key presses on any other key (apparently except `ctrl`) trigger the hotkey again.

Demo: https://stackblitz.com/edit/react-eqz5gk?file=src/App.js

In v3 this issue did not occur.

**Comment by JohannesKlauss** [comment]
This is a weird issue. Seems like there is a problem with attaching multiple event listeners and cleaning them up again, not sure though. I am investigating

**Comment by nosferatu500** [comment]
@JohannesKlauss 

> if you trigger the hotkey but do not release meta, any subsequent key presses on any other key (apparently except ctrl) trigger the hotkey again.

same here. in my case, pressing `meta+c` and while holding `meta` pressing `meta+v` will invoke `[meta+c, meta+c, meta+v]`. In v3 the behavior is different: `[meta+c, meta+v]`.

**Comment by JohannesKlauss** [comment]
This is an issue realted to macOS. I am trying to come up with a workaround for that

**Comment by nosferatu500** [comment]
Got it! Thank you!

**Comment by JohannesKlauss** [comment]
@pionxzh Turns out this is not a bug, but correct behavior. You are mutating the state and set that state variable as a dependency in the `useHotkeys` hook. So once it updates the state all internal key handlers will unbind and rebind meaning, that the keydown event for the right arrow key will not get handled.
Here is an example of how you should do this: https://stackblitz.com/edit/react-xffanw?devToolsHeight=33&file=package.json,src%2FApp.js

**Comment by pionxzh** [comment]
Thanks for pointing that out. But I can still reproduce case 2 in your updated version. Is it also an expected behavior?

**Comment by JohannesKlauss** [comment]
Does it happen in 4.1.0?

**Comment by pionxzh** [comment]
> Does it happen in 4.1.0?

I just tested 4.1.0, and it's good now. Thanks 🙏

**Comment by nosferatu500** [comment]
>same here. in my case, pressing meta+c and while holding meta pressing meta+v will invoke [meta+c, meta+c, meta+v]. In v3 the behavior is different: [meta+c, meta+v].

@JohannesKlauss

Seems to be fixed in `v4.3.0` but now I found another one issue:

pressing `meta+c` and while holding `meta` pressing `meta+v` **a few times** will invoke `[meta+c, meta+v]`. In v3 the behavior is different: `[meta+c, meta+v, meta+v, meta+v]`.
